genai_stack/memory/vectordb.py

```diff
@@ -19,42 +21,56 @@ class VectorDBMemoryConfig(BaseMemoryConfig):
 
 class VectorDBMemory(BaseMemory):
     config_class = VectorDBMemoryConfig
-    memory = None
     lc_client = None
 
     def _post_init(self, *args, **kwargs):
         config:VectorDBMemoryConfigModel  = self.config.config_data
+        
+        self.kwarg_map = create_kwarg_map(config=config)
 
-        # We have to pass the index name in two places, one is to Vectordb and to 
-        # VectorStoreRetriever.
-        # in case of weaviate, if we pass the index name in lowercase, the weaviate will
-        # internally convert it to pascal for schema/collection
-        # eg passed index name => chatting, weaviate converted to Chatting,
-        # But if VectorStoreRetriever use the lowercased index name, 
-        # it throws index error, since the weaviate changed to pascal.
-        # To handle this we are converting the index name to pascal before intializing 
-        # the Vectordb and Vectorstoreretriever, and to 
-        # maintain the consistency, we are also converting the chromadb index name to 
-        # pascal, instead of conditionally doing only for weaviate.
-        kwarg_map, index_name = format_index_name(config=config)
-
-        self.lc_client = self.mediator.create_index(kwarg_map)
-
-        retriever = self.lc_client.as_retriever(
-            search_kwargs = {'k':4}
-        )
-
-        self.memory = VectorStoreRetrieverMemory(
-            retriever=retriever,
-            memory_key=index_name,
-            return_docs=True
-        )
+        self.lc_client = self.mediator.create_index(kwarg_map=self.kwarg_map)
 
     def add_text(self, user_text: str, model_text: str):
-        self.memory.save_context({"input": user_text}, {"output": model_text})
+        id = "7a476819-799d-4f0b-a57a-300cc10d130f"
 
-    def get_chat_history(self, query):
-        documents = self.memory.load_memory_variables({
-            "prompt": query
-        })[self.memory.memory_key]
-        return parse_chat_conversation_history_search_result(search_results=documents)
\ No newline at end of file
+        _kwarg_map = {**self.kwarg_map}
+
+        # Storing a random embeddings
+        _kwarg_map['ChromaDB']['embeddings'] = [1,2,3,4]
```

Comment:
Is there no alternative other than hardcoding the embedding ?

Reply:
we cant pass none. we can create embeddings using embedding component i guess not sure, it make sense to create embedding when we do the update too not just at the time of creating. if we only want when creating not updating then in that case we can pass this random hardcode embedding

Reply:
we cant pass none. we can create embeddings using embedding component i guess not sure, it make sense to create embedding when we do the update too not just at the time of creating. if we only want when creating not updating then in that case we can pass this random hardcode embedding